app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/FurtherReturnGuidanceController.scala

```diff
@@ -0,0 +1,134 @@
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.returns.triage
+
+import com.google.inject.Inject
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents, Result}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.config.{ErrorHandler, ViewConfig}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.returns
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.SessionUpdates
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.actions.{AuthenticatedAction, RequestWithSessionData, SessionDataAction, WithAuthAndSessionDataAction}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.JourneyStatus.{FillingOutReturn, StartingNewDraftReturn, StartingToAmendReturn}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.SessionData
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.RepresentativeType
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.{Logging, toFuture}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.views
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import scala.concurrent.Future
+
+class FurtherReturnGuidanceController @Inject() (
+  errorHandler: ErrorHandler,
+  val authenticatedAction: AuthenticatedAction,
+  val sessionDataAction: SessionDataAction,
+  cc: MessagesControllerComponents,
+  guidancePage: views.html.returns.triage.further_return_guidance
+)(implicit viewConfig: ViewConfig)
+    extends FrontendController(cc)
+    with WithAuthAndSessionDataAction
+    with SessionUpdates
+    with Logging {
+
+  import FurtherReturnGuidanceController._
+
+  def guidance(back: String): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withState(request) { (_, state) =>
+        backLinkMappings.get(back) match {
+          case None           =>
+            logger.warn(s"Could not find back link location for '$back'")
+            errorHandler.errorResult()
+
+          case Some(backLink) =>
+            Ok(
+              guidancePage(
+                backLink,
+                state.fold(
+                  _.fold(
+                    _.subscribedDetails.isATrust,
+                    _.subscribedDetails.isATrust
+                  ),
+                  _.subscribedDetails.isATrust
+                ),
+                getRepresentativeType(state)
+              )
+            )
+        }
+
+      }
+    }
+
+  def getRepresentativeType(
+    state: Either[Either[StartingToAmendReturn, StartingNewDraftReturn], FillingOutReturn]
+  ): Option[RepresentativeType] =
+    state
+      .fold(
+        _.fold(
+          _ => None,
+          _.newReturnTriageAnswers.fold(
+            _.representativeType(),
+            _.representativeType()
+          )
+        ),
+        _.draftReturn.representativeType()
+      )
+
+  private def withState(request: RequestWithSessionData[_])(
```

Comment:
could you rename this to `withJourneyState` as that makes it more clear what state it is referring to